### ISSUE TITLE
kubernetes-deploy fails with API version 1.16

### ISSUE THREAD

**Issue by redmcg** [Shopify/kubernetes-deploy#592]
# Bug report

`kubernetes-deploy` (and `krane deploy`) fail during the prune phase on deploy to a k8s cluster using v1.16 of the api. This is because the following apis:
```
extensions/v1beta1/ReplicaSet
extensions/v1beta1/DaemonSet
extensions/v1beta1/Deployment
apps/v1beta1/StatefulSet
```

have been deprecated. See https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/ for reference.

I have created a fork that addresses this issue for v1.16 (see https://github.com/redmcg/kubernetes-deploy/tree/fixforv1.16), but I did not submit this as a PR as I'm worried it will break older versions of the k8s API (where the new APIs may not be implemented).

**Expected behavior:**
`kubernetes-deploy` (and `krane deploy`) to successfully deploy to a k8s v1.16 cluster

**Actual behavior:**
`kubernetes-deploy` (and `krane deploy`) fail during the prune phase with the following error:
```
[INFO][2019-10-18 11:19:43 +1100]	------------------------------------------Result: FAILURE-------------------------------------------
[FATAL][2019-10-18 11:19:43 +1100]	Command failed: apply -f /tmp/d20191018-14256-1x9qqjf --prune --all --prune-whitelist\=core/v1/configmap --prune-whitelist\=core/v1/pod --prune-whitelist\=core/v1/service --prune-whitelist\=core/v1/resourcequota --prune-whitelist\=core/v1/secret --prune-whitelist\=core/v1/serviceaccount --prune-whitelist\=core/v1/podtemplate --prune-whitelist\=batch/v1/job --prune-whitelist\=extensions/v1beta1/replicaset --prune-whitelist\=extensions/v1beta1/daemonset --prune-whitelist\=extensions/v1beta1/deployment --prune-whitelist\=extensions/v1beta1/ingress --prune-whitelist\=networking.k8s.io/v1/networkpolicy --prune-whitelist\=apps/v1beta1/statefulset --prune-whitelist\=autoscaling/v1/horizontalpodautoscaler --prune-whitelist\=policy/v1beta1/poddisruptionbudget --prune-whitelist\=batch/v1beta1/cronjob --prune-whitelist\=rbac.authorization.k8s.io/v1/role --prune-whitelist\=rbac.authorization.k8s.io/v1/rolebinding
[FATAL][2019-10-18 11:19:43 +1100]	
[FATAL][2019-10-18 11:19:43 +1100]	WARNING: Any resources not mentioned in the error(s) below were likely created/updated. You may wish to roll back this deploy.
[FATAL][2019-10-18 11:19:43 +1100]	
[FATAL][2019-10-18 11:19:43 +1100]	Unidentified error(s):
[FATAL][2019-10-18 11:19:43 +1100]	    error: no matches for kind "ReplicaSet" in version "extensions/v1beta1"

```

**Version(s) affected:**
v0.29.0 (but both `kubernetes-deploy` and `krane deploy` failed with my own builds from GitHub master)

### Steps to Reproduce

1. Deploy a v1.16 k8s cluster (e.g. `snap install microk8s --classic`)
2. Attempt a deploy to this cluster

**Comment by dturn** [comment]
Thanks for the bug report. We've merged your changes into master and are going to cut a release shortly.  We're aiming to provide official 1.16 support soon.

**Comment by redmcg** [comment]
I just noticed this is fixed in v0.30. So for my purposes this issue is resolved.

**Comment by dturn** [comment]
Glad its working for you. As an fyi we're adding formal 1.16 support in https://github.com/Shopify/kubernetes-deploy/pull/598.

**Comment by douglas** [comment]
Hello @redmcg! Just merged #598 in `master` in case you want to test it 🎉 

I will close this issue for now, but feel free to open another issue if you bump into any problems! Thanks!

**Comment by redmcg** [comment]
Thanks all. I'll check it out
